ko/news/_posts/2018-05-31-ruby-2-6-0-preview2-released.md

```diff
@@ -0,0 +1,121 @@
+---
+layout: news_post
+title: "루비 2.6.0-preview2 릴리스"
+author: "naruse"
+translator: "shia"
+date: 2018-05-31 00:00:00 +0000
+lang: ko
+---
+
+루비 2.6.0-preview2 릴리스를 알리게 되어 기쁩니다.
+
+루비 2.6.0-preview2는 루비 2.6.0의 두번째 프리뷰입니다.
+이 프리뷰는 JIT이라는 중요한 새 기능을 포함하고 있기 때문에 평소보다 빠르게 릴리스됩니다.
+그 일부를 아래에서 소개합니다.
+
+## JIT
+
+루비 2.6은 JIT(Just-in-time) 컴파일러의 첫 구현체를 포함합니다.
+
+JIP 컴파일러는 루비 프로그램의 실행 성능을 향상시키는 것이 목적입니다.
+다른 언어의 일반적인 JIT 컴파일러와는 다르게, 루비의 JIT 컴파일러는 C 코드를 디스크에 출력한 뒤, 일반적인 C 컴파일러 프로세스를 사용해 네이티브 코드를 생성하도록 합니다.
+다음을 참고하세요. [Vladimir Makarov가 작성한 MJIT 구조](https://github.com/vnmakarov/ruby/tree/rtl_mjit_branch#mjit-organization).
+
+JIT 컴파일을 사용하려면 `--jit` 옵션을 커맨드라인이나 `$RUBYOPT` 환경 변수에 지정합니다. `--jit-verbose=1`을 지정하면 실행 중인 JIT 컴파일의 기본적인 정보를 출력합니다. 다른 옵션에 대해서는 `ruby --help`를 확인하세요.
+
+이번 JIT 릴리스의 주 목적은 2.6 릴리스 전에 각 플랫폼에서 잘 동작하는지, 보안상의 문제가 발생하는지 미리 확인하는 것입니다.
+현재 JIT 컴파일러는 루비가 gcc나 clang로 빌드되었으며, 해당 컴파일러가 런타임에서 사용 가능한 경우에만 이용할 수 있습니다. 그 이외에는 아직 이용할 수 없습니다.
+
+2.6.0-preview2에서는 JIT를 이용하기 위한 기반을 추가하고 매우 적은 최적화만이 구현되어 있습니다. 이 릴리스에 대해서 간단한 벤치마크를 이용하여 잠재적인 성능 향상을 확인할 수 있습니다만, Rails 애플리케이션과 같은 대규모 프로그램에서 루비 JIT 컴파일러의 최종적인 성능을 측정할만한 단계는 아닙니다.
+
+앞으로 루비의 성능을 한 단계 향상시켜줄 것으로 기대되고 있는 메소드 인라인화를 구현할 예정입니다.
+또한 지원 환경을 늘려나갈 계획이며, 그 첫 번째로는 Visual Studio를 지원하는 것입니다.
+
+새로운 루비의 성능을 기대해주세요.
+
+## RubyVM::AST [Experimental]
+
+루비 2.6에는 `RubyVM::AST` 모듈이 도입되었습니다.
+
+이 모듈에은 문자열을 파싱하여 AST(추상구문트리)의 Node를 돌려주는 `parse` 메소드, 파일을 파싱하는 `parse_file` 메소드가 들어있습니다.
+`RubyVM::AST::Node`도 도입되었습니다. 이 클래스의 인스턴스로부터 위치정보나 자식 노드를 얻을 수 있습니다. 이 기능은 실험적으로 포함되었으며, 호환성을 보장하지 않습니다.
+
+## 새로운 기능
+
+* `Kernel#yield_self`의 별칭으로 `then` 이 추가되었습니다. [[Feature #14594]](https://bugs.ruby-lang.org/issues/14594)
+
+* `rescue` 가 없는 `else` 가 문법 에러가 됩니다. [EXPERIMENTAL]
+
+* ASCII 이외의 대문자라도 상수를 정의할 수 있게 됩니다. [[Feature #13770]](https://bugs.ruby-lang.org/issues/13770)
```

Comment:
I think it's about the starting letter. On Ruby 2.5.1, this doesn't cause an error:

``` ruby
>> ÃA = 1
>> ÃA = 2
```

But this causes an error:
``` ruby
>> AÃ = 1
>> AÃ = 2
(irb):2: warning: already initialized constant AÃ
(irb):1: warning: previous definition of AÃ was here
```

Reply:
you're right. 👍 This is just my mistake.